From 2198234de783fed27b94da344bb29e1d245b9f6f Mon Sep 17 00:00:00 2001
From: Stephane Danieau <stephane.danieau@st.com>
Date: Wed, 8 Jan 2025 18:04:36 +0100
Subject: [PATCH] arm64: defconfig: switch drivers from built-in to module

Switch as much as possible drivers in module configuration
keeping only the backbone drivers in built-in to avoid to slow
down the kernel start.

Change-Id: I32731c117af8eb38366d166b92bdbfb5001959d5
Signed-off-by: Stephane Danieau <stephane.danieau@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/427698
Tested-by: Stephane DANIEAU <stephane.danieau@foss.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Stephane DANIEAU <stephane.danieau@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 .../fragment-02-defconfig-addons.config       | 41 +++++++++++--------
 1 file changed, 23 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/configs/fragment-02-defconfig-addons.config b/arch/arm64/configs/fragment-02-defconfig-addons.config
index ba2f06db3cf4..f4809c68ce7b 100644
--- a/arch/arm64/configs/fragment-02-defconfig-addons.config
+++ b/arch/arm64/configs/fragment-02-defconfig-addons.config
@@ -11,6 +11,7 @@ CONFIG_CORESIGHT_SINK_TPIU=y
 CONFIG_CORESIGHT_SOURCE_ETM4X=y
 CONFIG_CORESIGHT_STM=y
 CONFIG_COUNTER=m
+CONFIG_CPUFREQ_DT=m
 CONFIG_CPU_IDLE_GOV_TEO=y
 CONFIG_CRYPTO_DEV_STM32_CRC=m
 CONFIG_CRYPTO_DEV_STM32_CRYP=m
@@ -27,24 +28,24 @@ CONFIG_DRM_PANEL_BRIDGE=y
 CONFIG_DRM_PANEL_LVDS=y
 CONFIG_DRM_PANEL_RAYDIUM_RM68200=y
 CONFIG_DRM_PANEL_SIMPLE=y
-CONFIG_DRM_SIMPLEDRM=y
-CONFIG_DRM_STM=y
-CONFIG_DRM_STM_DSI=y
-CONFIG_DRM_STM_LVDS=y
+CONFIG_DRM_SIMPLEDRM=m
+CONFIG_DRM_STM=m
+CONFIG_DRM_STM_DSI=m
+CONFIG_DRM_STM_LVDS=m
 CONFIG_DWMAC_GENERIC=y
 CONFIG_DWMAC_STM32=y
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_GENERIC_IRQ_DEBUGFS=y
 CONFIG_HDMI=y
 CONFIG_I2C_RPMSG=m
-CONFIG_I2C_STM32F7=y
+CONFIG_I2C_STM32F7=m
 CONFIG_I3C=y
 CONFIG_IGC=m
 CONFIG_IIO_RESCALE=m
 CONFIG_IIO_STM32_LPTIMER_TRIGGER=m
 CONFIG_IIO_STM32_TIMER_TRIGGER=m
 CONFIG_JFFS2_FS=y
-CONFIG_MAILBOX_CDEV=y
+CONFIG_MAILBOX_CDEV=m
 CONFIG_MEDIA_TEST_SUPPORT=y
 CONFIG_MFD_STM32_LPTIMER=m
 CONFIG_MFD_STM32_TIMERS=m
@@ -65,10 +66,11 @@ CONFIG_NVMEM_STM32_ROMEM=y
 CONFIG_NVMEM_STM32_TAMP=y
 CONFIG_OF_OVERLAY=y
 CONFIG_PCIEASPM_POWERSAVE=y
-CONFIG_PCIE_STM32=y
-CONFIG_PCIE_STM32_EP=y
-CONFIG_PHY_STM32_COMBOPHY=y
-CONFIG_PHY_STM32_USB2PHY=y
+CONFIG_PCIE_STM32=m
+CONFIG_PCIE_STM32_EP=m
+CONFIG_PHY_STM32_COMBOPHY=m
+CONFIG_PHY_STM32_USB2PHY=m
+CONFIG_PINCTRL_STM32_HDP=m
 CONFIG_PTM=y
 CONFIG_PWM_STM32=m
 CONFIG_PWM_STM32_LP=m
@@ -79,12 +81,12 @@ CONFIG_RPMSG_CTRL=m
 CONFIG_RPMSG_IRQ=m
 CONFIG_RPMSG_TTY=m
 CONFIG_RPMSG_VIRTIO=m
-CONFIG_RTC_DRV_STM32=y
+CONFIG_RTC_DRV_STM32=m
 CONFIG_SAMPLES=y
 CONFIG_SAMPLE_RPMSG_CLIENT=m
 CONFIG_SCMI_STM32MP_OSTL_V5=y
 CONFIG_SD_ADC_MODULATOR=m
-CONFIG_SENSORS_MR75203=y
+CONFIG_SENSORS_MR75203=m
 CONFIG_SMSC_PHY=y
 CONFIG_SND_SOC_STM32_I2S=m
 CONFIG_SND_SOC_STM32_MDF=m
@@ -102,15 +104,17 @@ CONFIG_STM32_HYPERBUS=y
 CONFIG_STM32_I3C_MASTER=m
 CONFIG_STM32_IPCC=y
 CONFIG_STM32_LPTIMER_CNT=m
-CONFIG_STM32_M0_RPROC=y
+CONFIG_STM32_M0_RPROC=m
 CONFIG_STM32_MDF_ADC=m
 CONFIG_STM32_MDF_CORE=m
 CONFIG_STM32_OMI=y
 CONFIG_STM32_OMM=y
-CONFIG_STM32_RPROC=y
+CONFIG_STM32_RPROC=m
 CONFIG_STM32_TIMER_CNT=m
-CONFIG_STMMAC_ETH=y
-CONFIG_STMMAC_PLATFORM=y
+CONFIG_STM32_WATCHDOG=m
+CONFIG_STMMAC_ETH=m
+CONFIG_STMMAC_PLATFORM=m
+CONFIG_SYSCON_REBOOT_MODE=m
 CONFIG_TEE_REMOTEPROC=y
 CONFIG_TOUCHSCREEN_GOODIX=y
 CONFIG_TOUCHSCREEN_ILI210X=y
@@ -118,10 +122,11 @@ CONFIG_TYPEC_UCSI=m
 CONFIG_UBIFS_FS=y
 CONFIG_UCSI_STM32G0=m
 CONFIG_USB_CONFIGFS_F_UVC=y
-CONFIG_USB_DWC3_STM32=y
+CONFIG_USB_DWC2=m
+CONFIG_USB_DWC3_STM32=m
 CONFIG_USB_HIDDEV=y
 CONFIG_USB_ONBOARD_HUB=y
-CONFIG_USB_STM32_USBH=y
+CONFIG_USB_STM32_USBH=m
 CONFIG_V4L_TEST_DRIVERS=y
 CONFIG_VIDEO_IMX335=m
 CONFIG_VIDEO_STM32_CSI=m
-- 
2.50.1

